### PR TITLE
feat(openapi): round-trip golden harness + OpenAPI path templating (PR1)

### PR DESCRIPTION
**Pull request by gaborage**
## PR 1 of the OpenAPI-tool fidelity roadmap

The `go-bricks-openapi` generator builds, vets, passes all tests, and sits at 90.2% coverage — yet it emits **valid-but-empty specs** for idiomatic services (handlers on a separate `*Handler` struct lose all schemas; `Result[R]` responses are dropped; Echo `:id` paths aren't templated; etc.). Those are all *silent* failures no unit test catches, because **no test runs `AnalyzeProject → Generate` end-to-end**.

This PR lays the foundation for the rest of the roadmap and ships the first real correctness fix.

### What's in it

**1. Round-trip golden + validation harness (`internal/spectest`)**
- Runs the real pipeline (`analyzer.AnalyzeProject` → `generator.Generate`) over `testdata/<case>/` fixtures.
- Validates the emitted document as **OpenAPI 3.0 in-process via kin-openapi** — deterministic, no network/toolchain. This is the primary gate.
- Golden-compares the exact emitted YAML (refresh with `go test ./internal/spectest -update`).
- This is the regression net every later fidelity PR relies on: a fix shows up as a reviewable golden diff; a structural regression fails `Validate`.

**2. OpenAPI path templating (`analyzer.normalizePath`)**
- Converts Echo route params `:id` → `{id}` so parameterized paths are valid OpenAPI 3.0 and match the declared path parameters (previously `/users/:id` was emitted verbatim — invalid templating).
- Catch-all wildcards (`/files/*`) are intentionally left **literal** for now: templating them requires a synthesized path parameter, which is deferred to the parameter-fidelity work. Leaving them literal matches today's behavior (no regression).

**3. CI structural-validation gate**
- `make validate-spec` generates a fixture spec and lints it with **pinned redocly 2.31.5**, wired into the Ubuntu `tool-build` job.
- `redocly.yaml` extends `recommended` and relaxes **only** the rules for features the generator doesn't emit yet (`no-empty-servers`, `security-defined`, `no-unused-components`, `info-license`) — each annotated with the PR that re-enables it, so the gate tightens over time without ever blocking a self-contained PR.

### Coverage
`normalizePath` 100%, `spectest` package 85.7%; tool total stays **90.2%**. `make test`/`test-coverage` now include `./internal/spectest/...` so the harness runs under `make check` locally, not only in CI.

### Pre-push verification
- `make check` (fmt + lint + race tests incl. spectest + validate-cli) — green
- `make validate-spec` (redocly) — clean
- `/code-review` — run; findings applied (critical: golden/redocly files were gitignored under the allowlist `.gitignore` and now allowlisted; wildcard-templating regression removed; a silently-skipped test assertion re-armed)
- `/security-audit` — gosec 0 issues, govulncheck clean (incl. new kin-openapi deps), no secrets

### Notes
- `.gitignore`: allowlisted `**/testdata/**/*.yaml` (goldens) and `tools/openapi/redocly.yaml` — the repo uses an ignore-everything allowlist, so new file types need explicit un-ignoring.
- Adds `github.com/getkin/kin-openapi` as a (test-exercised) dependency of the tool module.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **New Features**
  * CI now runs an Ubuntu-only validation step to lint/validate generated OpenAPI specs.
  * Route path parameters are normalized to OpenAPI template form (e.g., /users/{id}).

* **Tests**
  * Added end-to-end spec generation and validation tests with golden fixtures and edge-case coverage.

* **Chores**
  * Added Redocly lint configuration; updated ignore/gitattributes and ancillary configs for stable spec testing.

[Review Change Stack](https://app.coderabbit.ai/change-stack/gaborage/go-bricks/pull/484?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

### CODE REVIEW

File: tools/openapi/internal/spectest/testdata/path_param/expected.yaml [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+      type: object
+      properties:
+        iD:
+          type: integer
+          format: int64
+        page:
+          type: integer
+          format: int32
+      required:
+        - iD
```
[the remark this answers is not in the document]
Comment (gaborage) [reply tools/openapi/internal/spectest/testdata/path_param/expected.yaml:62]: Good eye — `iD` is the symptom of two **pre-existing generator bugs** the golden intentionally captures so they show up as a reviewable diff when fixed:

1. `GetProductReq` is a *params-only* struct (all fields are `param:`/`query:`) and should not be emitted as a `components` schema at all.
2. When a field has no `json:` tag, the generator derives the property name by lower-casing only the first letter (`ID` → `iD`) instead of applying proper JSON casing.

Both are tracked in the roadmap gap inventory and fixed in a later PR (params-only-component gating + schema-fidelity); fixing them here would be out of scope for PR1 (harness + path templating). The OpenAPI **parameter** name is correctly `id` (from the `param:"id"` tag) — only the spurious component shows `iD`. Leaving the golden as-is on purpose.
